cmd/exo/upgrade.go

```diff
@@ -46,7 +50,17 @@ var upgradeCmd = &cobra.Command{
 		case -1:
 			fmt.Println("Upgrade needed")
 			// TODO: Prompt for confirmation?
-			return upgrade.UpgradeSelf()
+			statePath := filepath.Join(cfg.VarDir, "state.json")
+			deviceIDPath := filepath.Join(cfg.VarDir, "deviceid")
+			store := statefile.New(statefile.Config{
+				StoreFilename:    statePath,
+				DeviceIDFilename: deviceIDPath,
+			})
+			ensureDeviceOut, err := store.EnsureDevice(ctx, &state.EnsureDeviceInput{})
```

Comment:
not sure i love the idea of the cli directly accessing the store api, since the intent is that the daemon manages this state

Reply:
How about if I make a kernel endpoint that returns it, then have this entrypoint call `ensureDaemon()` and make an API call?

Reply:
I think that the less the upgrade path relies on the daemon, the better. Maybe we ship some bug where the daemon is crashing or something & we want people to be able to upgrade to fix it!

Reply:
to be clear: I think it's OK that the daemon and the CLI share code to do this, just not specifically the statefile package, since that's for managing the state of components/stacks. A package dedicated to the subset of functionality for device ids, upgrades, etc seems like a more minimal surface for bugs.